testbed/correctnesstests/utils.go

```diff
@@ -44,6 +45,64 @@ func CreateConfigYaml(
 		}
 	}
 
+	var pipeline1 string
+	switch sender.(type) {
+	case testbed.TraceDataSender:
+		pipeline1 = "traces"
+	case testbed.MetricDataSender:
+		pipeline1 = "metrics"
+	case testbed.LogDataSender:
+		pipeline1 = "logs"
+	default:
+		t.Error("Invalid DataSender type")
+	}
+
+	if connector != nil {
+		pipeline2 := connector.GetReceiverType()
+
+		format := `
+receivers:%v
+exporters:%v
+processors:
+  %s
+
+extensions:
+
+connectors:%v
+
+service:
+  telemetry:
+    metrics:
+      address: 127.0.0.1:%d
+    logs:
+      level: "debug"
+  extensions:
+  pipelines:
+    %s/in:
+      receivers: [%v]
+      processors: [%s]
+      exporters: [%v]
+    %s/out:
+      receivers: [%v]
+      exporters: [%v]
+`
+		return fmt.Sprintf(
+			format,
+			sender.GenConfigYAMLStr(),
+			receiver.GenConfigYAMLStr(),
```

Comment:
Same thing here, why is this named `receiver` but is being added to the `exporters` section?

Reply:
The reasoning is the same as above